src/main/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtils.java

```diff
@@ -34,18 +39,59 @@
      * If both the {@link CartDiscount} and the {@link CartDiscountDraft} have the same cart discount value,
      * then no update action is needed and hence an empty {@link Optional} is returned.
      *
+     * <p>Note: Order is not considered as an update action for the {@link AbsoluteCartDiscountValue}s
+     *
      * @param oldCartDiscount the cart discount which should be updated.
      * @param newCartDiscount the cart discount draft where we get the new cart discount value.
      * @return A filled optional with the update action or an empty optional if the cart discount values are identical.
      */
     @Nonnull
     public static Optional<UpdateAction<CartDiscount>> buildChangeValueUpdateAction(
-        @Nonnull final CartDiscount oldCartDiscount,
-        @Nonnull final CartDiscountDraft newCartDiscount) {
+            @Nonnull final CartDiscount oldCartDiscount,
+            @Nonnull final CartDiscountDraft newCartDiscount) {
+
+        if (oldCartDiscount.getValue() instanceof AbsoluteCartDiscountValue
+                && newCartDiscount.getValue() instanceof AbsoluteCartDiscountValue) {
+
+            final List<MonetaryAmount> amountsFromCartDiscount =
+                    ((AbsoluteCartDiscountValue) oldCartDiscount.getValue()).getMoney();
+            final List<MonetaryAmount> amountsFromCartDiscountDraft =
+                    ((AbsoluteCartDiscountValue) newCartDiscount.getValue()).getMoney();
+
+            return buildChangeAbsoluteValueUpdateAction(amountsFromCartDiscount, amountsFromCartDiscountDraft);
+        }
+
         return buildUpdateAction(oldCartDiscount.getValue(), newCartDiscount.getValue(),
             () -> ChangeValue.of(newCartDiscount.getValue()));
     }
 
+    @Nonnull
+    private static Optional<UpdateAction<CartDiscount>> buildChangeAbsoluteValueUpdateAction(
+            @Nonnull final List<MonetaryAmount> amountsFromCartDiscount,
+            @Nonnull final List<MonetaryAmount> amountsFromCartDiscountDraft) {
+
+        final List<MonetaryAmount> newAmounts = new ArrayList<>();
+        boolean isAmountNew;
+        for (MonetaryAmount amountDraft : amountsFromCartDiscountDraft) {
+            isAmountNew = true;
+            for (MonetaryAmount amount : amountsFromCartDiscount) {
```

Comment:
maybe a more efficient implementation than `n^2` implementation is to: 
- Check if they have different size first, 
   - if yes, then build action. 
   - if not, then 1. sort then 2. use equals for comparison.

refernece: https://stackoverflow.com/a/13501200/5654504

Reply:
is the complexity important for this algorithm because N won't be big enough for performance? 

they could be in different sizes but could be the same elements, the draft could have nullable values or duplicate values.

Reply:
> they could be in different sizes but could be the same elements, the draft could have nullable values or duplicate values.

actually duplicate (and null) values are not allowed by the platform (I have just tried it now):
```json
"code": "InvalidOperation",
"message": "Duplicate currencies: [EUR]"
```

So If someone is trying to sync in duplicate money values, then (s)he will get this error message from the platform. The sync library is not responsible/expected to remove those duplicate or null values. 

> is the complexity important for this algorithm because N won't be big enough for performance?

I would say the complexity here is very critical, since I think there should not be so many `Money` Objects in the list. **But I only _think_.** So why not just have the better performant one? 

Having efficient code gives a good impression about the code base in general to any coder. Which in turn leads to maintaining it also with quality.  Especially, that is self-understandable and not confusing and I think this better performant one is quite clear to understand.

Reply:
Efficient code is related to the N also related to the memory usage ext, anyway I will check this StackOverflow answer if it's more understandable, why not :)

Reply:
But this algorithm just checks the equality, we need to find the diff of the two lists too, check this test : https://github.com/commercetools/commercetools-sync-java/blob/5bd6ba4cf38ed2c70670271c091ecc3fa61e76f5/src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java#L285

Anway, let's discuss it together 😄